File: views/src/main/java/dlgdev/weighttracker/dagger/DateFormatModule.java
package dlgdev.weighttracker.dagger;

import android.content.Context;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Locale;

import dagger.Module;
import dagger.Provides;

@Module
public class DateFormatModule {

	private Context context;

	public DateFormatModule(Context context) {
		this.context = context;
	}

	@Provides Locale provideLocale() {
		return context.getResources().getConfiguration().locale;
	}

	@Provides DateFormat provideDateFormat(Locale locale) {
		return new SimpleDateFormat("yyyy-MM-dd - HH:mm:ss", locale);
	}
}

File: views/src/main/java/dlgdev/weighttracker/dagger/DateFormatModule.kt
package dlgdev.weighttracker.dagger

import android.content.Context
import dagger.Module
import dagger.Provides
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

@Module
class DateFormatModule(private val context: Context) {

    @Provides internal fun provideLocale(): Locale {
        return context.resources.configuration.locale
    }

    @Provides internal fun provideDateFormat(locale: Locale): DateFormat {
        return SimpleDateFormat("yyyy-MM-dd - HH:mm:ss", locale)
    }
}
